package/MDAnalysis/analysis/density.py

```diff
@@ -408,8 +417,15 @@ def _prepare(self):
             # ideal solution would use images: implement 'looking across the
             # periodic boundaries' but that gets complicated when the box
             # rotates due to RMS fitting.
-            smin = np.min(coord, axis=0) - self._padding
-            smax = np.max(coord, axis=0) + self._padding
+            try:
+                smin = np.min(coord, axis=0)
+                smax = np.max(coord, axis=0)
+            except ValueError as err:
+                errmsg = ("No atoms in selection at current frame"
+                          " and no user defined grid")
+                raise ValueError(errmsg) from err
+            smin = smin - self._padding
```

Comment:
why is this not being done on line 421 and 422 respectively as per the old code? Either smin/smax are going to be defined or you're going to get the ValueError?

Reply:
My bad, that's a unnecessary step. I'll remove it